File: voxtral_realtime/windows/VoxtralRealtime/VoxtralRealtime/Models/Snippet.cs
// Copyright (c) Meta Platforms, Inc. and affiliates.
// All rights reserved.
// This source code is licensed under the BSD-style license found in the
// LICENSE file in the root directory of this source tree.

using System.Text.Json.Serialization;

namespace VoxtralRealtime.Models;

public class Snippet
{
    [JsonPropertyName("id")]
    public Guid Id { get; set; } = Guid.NewGuid();

    [JsonPropertyName("name")]
    public string Name { get; set; } = "";

    [JsonPropertyName("trigger")]
    public string Trigger { get; set; } = "";

    [JsonPropertyName("content")]
    public string Content { get; set; } = "";

    [JsonPropertyName("isEnabled")]
    public bool IsEnabled { get; set; } = true;

    [JsonPropertyName("notes")]
    public string? Notes { get; set; }

    [JsonPropertyName("lastUsedAt")]
    public DateTime? LastUsedAt { get; set; }

    public Snippet Clone() => new()
    {
        Id = Id,
        Name = Name,
        Trigger = Trigger,
        Content = Content,
        IsEnabled = IsEnabled,
        Notes = Notes,
        LastUsedAt = LastUsedAt
    };
}

File: voxtral_realtime/macos/VoxtralRealtime/Models/Snippet.swift
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree.
 */

import Foundation

struct Snippet: Identifiable, Codable, Sendable, Hashable {
    let id: UUID
    var name: String
    var trigger: String
    var content: String
    var isEnabled: Bool
    var notes: String
    var lastUsedAt: Date?

    init(
        id: UUID = UUID(),
        name: String = "",
        trigger: String = "",
        content: String = "",
        isEnabled: Bool = true,
        notes: String = "",
        lastUsedAt: Date? = nil
    ) {
        self.id = id
        self.name = name
        self.trigger = trigger
        self.content = content
        self.isEnabled = isEnabled
        self.notes = notes
        self.lastUsedAt = lastUsedAt
    }
}
